extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,676 @@
+
+  /**
+   * Touch start
+   * @param {Event} e
+   */
+  onTouchStart_(e) {
+    e.stopPropagation();
+    this.pointerMoveX_(e.touches[0].pageX);
+
+    // In case, clear up remnants
+    this.unlisten_(this.unlistenTouchMove_);
+    this.unlisten_(this.unlistenTouchEnd_);
+
+    this.unlistenTouchMove_ =
+        listen(this.win, 'touchmove', this.onTouchMove_.bind(this));
+    this.unlistenTouchEnd_ =
+        listen(this.win, 'touchend', this.onTouchEnd_.bind(this));
+
+    this.resetHintInterval_(true);
+  }
+
+  /**
+   * Touch move
+   * @param {Event} e
+   */
+  onTouchMove_(e) {
+    e.stopPropagation();
+    this.pointerMoveX_(e.touches[0].pageX);
+  }
+
+  /**
+   * Touch end
+   * @param {Event} e
+   */
+  onTouchEnd_(e) {
+    e.stopPropagation();
+    this.unlisten_(this.unlistenTouchMove_);
+    this.unlisten_(this.unlistenTouchEnd_);
+
+    this.resetHintInterval_(!this.shouldHintLoop_);
+  }
+
+  /**
+   * On key down
+   * @param {Event} e
+   */
+  onKeyDown_(e) {
+    // Check if current element has focus
+    if (this.doc_.activeElement !== this.element) {
+      return;
+    }
+
+    this.resetHintInterval_(!this.shouldHintLoop_);
+
+    switch (e.key.toLowerCase()) {
+      case 'arrowup':
+      case 'arrowleft':
+        this.stepLeft_();
+        break;
+      case 'arrowdown':
+      case 'arrowright':
+        this.stepRight_();
+        break;
+      case 'pageup':
+        // prevent scrolling the page
+        e.preventDefault();
+        e.stopPropagation();
+        this.stepLeft_(true);
+        break;
+      case 'pagedown':
+        e.preventDefault();
+        e.stopPropagation();
+        this.stepRight_(true);
+        break;
+      case 'home':
+        e.preventDefault();
+        e.stopPropagation();
+        this.stepExactCenter_();
+        break;
+    }
+  }
+
+  /**
+   * Unlisten a listener and clear. If null, does nothing
+   * @param {UnlistenDef|null} unlistenHandle
+   * @private
+   */
+  unlisten_(unlistenHandle) {
+    if (unlistenHandle) {
+      unlistenHandle();
+      unlistenHandle = null;
+    }
+  }
+
+  /**
+   * Register events
+   */
+  registerEvents_() {
+    this.unlistenMouseDown_ =
+        listen(this.element, 'mousedown', this.onMouseDown_.bind(this));
+    this.unlistenTouchStart_ =
+        listen(this.element, 'touchstart', this.onTouchStart_.bind(this));
+    this.unlistenKeyDown_ =
+        listen(this.element, 'keydown', this.onKeyDown_.bind(this));
+  }
+
+  /**
+   * Unregister events
+   */
+  unregisterEvents_() {
+    this.unlisten_(this.unlistenMouseDown_);
+    this.unlisten_(this.unlistenMouseMove_);
+    this.unlisten_(this.unlistenMouseUp_);
+
+    this.unlisten_(this.unlistenTouchStart_);
+    this.unlisten_(this.unlistenTouchMove_);
+    this.unlisten_(this.unlistenTouchEnd_);
+
+    this.unlisten_(this.unlistenKeyDown_);
+  }
+
+  /**
+   * Get current slider's percentage to the left
+   */
+  getCurrentSliderPercentage_() {
+    const {left: barLeft} =
```

Comment:
I think you can cache both of these values. Same comment as on 0.0 version above.